File: src/main/java/net/insprill/cjm/messages/types/TitleMessage.java
package net.insprill.cjm.messages.types;

import net.insprill.cjm.message.MessageVisibility;
import net.insprill.cjm.placeholder.Placeholders;
import net.insprill.xenlib.MinecraftVersion;
import net.insprill.xenlib.files.YamlFile;
import org.bukkit.entity.Player;

import java.io.File;
import java.util.List;

public class TitleMessage implements MessageType {

    private static final YamlFile config = new YamlFile("messages" + File.separator + "title.yml", false);

    @Override
    public YamlFile getConfig() {
        return config;
    }

    @Override
    public String getKey() {
        return "Messages";
    }

    @Override
    public String getName() {
        return "title";
    }

    @Override
    public void handle(Player primaryPlayer, List<Player> players, String rootPath, String chosenPath, MessageVisibility visibility) {
        String title = config.getString(chosenPath + ".Title");
        String subTitle = config.getString(chosenPath + ".SubTitle");
        int fadeIn = config.getInt(chosenPath + ".Fade-In");
        int stay = config.getInt(chosenPath + ".Stay");
        int fadeOut = config.getInt(chosenPath + ".Fade-Out");

        title = Placeholders.Companion.fillPlaceholders(primaryPlayer, title);
        subTitle = Placeholders.Companion.fillPlaceholders(primaryPlayer, subTitle);

        for (Player p : players) {
            sendTitle(p, title, subTitle, fadeIn, stay, fadeOut);
        }
    }

    @SuppressWarnings("deprecation")
    private void sendTitle(Player player, String title, String subTitle, int fadeIn, int stay, int fadeOut) {
        if (MinecraftVersion.isAtLeast(MinecraftVersion.v1_9_0)) {
            player.sendTitle(title, subTitle, fadeIn, stay, fadeOut);
        } else {
            player.sendTitle(title, subTitle);
        }
    }

}

File: src/main/kotlin/net/insprill/cjm/message/types/TitleMessage.kt
package net.insprill.cjm.message.types

import net.insprill.cjm.CustomJoinMessages
import net.insprill.cjm.extension.replacePlaceholders
import net.insprill.cjm.message.MessageVisibility
import net.md_5.bungee.api.chat.BaseComponent
import org.bukkit.entity.Player

class TitleMessage(private val plugin: CustomJoinMessages) : MessageType(plugin, "title", "Messages") {

    override fun handle(primaryPlayer: Player, recipients: List<Player>, chosenPath: String, visibility: MessageVisibility) {
        val titleStr = config.getString("$chosenPath.Title")?.replacePlaceholders(primaryPlayer) ?: ""
        val subTitleStr = config.getString("$chosenPath.SubTitle")?.replacePlaceholders(primaryPlayer) ?: ""
        if (titleStr.isBlank() && subTitleStr.isBlank()) return

        val title = BaseComponent.toLegacyText(*plugin.formatter.format(titleStr))
        val subTitle = BaseComponent.toLegacyText(*plugin.formatter.format(subTitleStr))
        val fadeIn = config.getInt("$chosenPath.Fade-In")
        val stay = config.getInt("$chosenPath.Stay")
        val fadeOut = config.getInt("$chosenPath.Fade-Out")

        for (player in recipients) {
            player.sendTitle(title, subTitle, fadeIn, stay, fadeOut)
        }
    }

}
